packages/dataviews/src/field-types/integer.tsx

```diff
@@ -0,0 +1,20 @@
+/**
+ * Internal dependencies
+ */
+import type { SortDirection } from '../types';
+
+function sort< Item >(
+	itemA: Item,
+	itemB: Item,
+	direction: SortDirection,
+	getValue: ( args: { item: Item } ) => any
```

Comment:
I've thinking a bit more here, I think we shouldn't pass `getValue` here at all, we should just pass `valueA` and `valueB` that are both integers. At this point (field type), there's no need to know about the item or the field or anything.

Reply:
I've pushed https://github.com/WordPress/gutenberg/pull/64064/commits/eaa0b63c8da763113d64aca0ec83342b14faa293

I like this more than previous approaches and it could scale well. For example, in validation, the field types such as integer would need to have access to `elements`, so we'd add that.

It's not entirely perfect, though: a generic field type such as `integer` doesn't know the "id" of the data it's working with so the `item` parameter there is useless.

Reply:
> Passing only the values is how I started. The trade-off with that approach is that it presumes the custom sort function wants to access data that's defined elsewhere, but it may want to access data that comes as part of the item (see author custom function).

This is only true for `field.sort` not for `fieldType.sort` where you always want to use the data.